src/scvi/external/methylvi/_model.py

```diff
@@ -0,0 +1,627 @@
+        return_numpy
+            Return a :class:`~numpy.ndarray` instead of a :class:`~pandas.DataFrame`.
+            DataFrame includes region names as columns. If either `n_samples=1` or
+            `return_mean=True`, defaults to `False`. Otherwise, it defaults to `True`.
+        importance_weighting_kwargs
+            Keyword arguments passed into
+            :meth:`~scvi.model.base.RNASeqMixin._get_importance_weights`.
+
+        Returns
+        -------
+        If `n_samples` is provided and `return_mean` is False,
+        this method returns a 3d tensor of shape (n_samples, n_cells, n_regions).
+        If `n_samples` is provided and `return_mean` is True, it returns a 2d tensor
+        of shape (n_cells, n_regions).
+        In this case, return type is :class:`~pandas.DataFrame` unless `return_numpy` is True.
+        Otherwise, the method expects `n_samples_overall` to be provided and returns a 2d tensor
+        of shape (n_samples_overall, n_regions).
+
+        If model was set up using a MuData object, a dictionary is returned with keys
+        corresponding to individual methylation modalities with values determined as
+        described above.
+        """
+        mdata = self._validate_anndata(mdata)
+
+        if indices is None:
+            indices = np.arange(mdata.n_obs)
+        if n_samples_overall is not None:
+            assert n_samples == 1  # default value
+            n_samples = n_samples_overall // len(indices) + 1
+        scdl = self._make_data_loader(adata=mdata, indices=indices, batch_size=batch_size)
+
+        region_mask = slice(None) if region_list is None else mdata.var_names.isin(region_list)
+
+        if n_samples > 1 and return_mean is False:
+            if return_numpy is False:
+                warnings.warn(
+                    "`return_numpy` must be `True` if `n_samples > 1` and `return_mean` "
+                    "is`False`, returning an `np.ndarray`.",
+                    UserWarning,
+                    stacklevel=settings.warnings_stacklevel,
+                )
+            return_numpy = True
+
+        store_distributions = weights == "importance"
+
+        exprs = defaultdict(list)
+        zs = []
+        qz_store = DistributionConcatenator()
+        px_store = DistributionConcatenator()
+        for tensors in scdl:
+            inference_kwargs = {"n_samples": n_samples}
+            inference_outputs, generative_outputs = self.module.forward(
+                tensors=tensors,
+                inference_kwargs=inference_kwargs,
+                generative_kwargs={},
+                compute_loss=False,
+            )
+
+            for modality in self.modalities:
+                exp_ = generative_outputs["px_mu"][modality]
+                exp_ = exp_[..., region_mask]
+                exprs[modality].append(exp_.cpu())
+            if store_distributions:
+                qz_store.store_distribution(inference_outputs["qz"])
```

Comment:
Are qz_store and px_store for all reasonable cases of modalities?

Reply:
Thanks for the catch- fixed! I hadn't properly accounted for different methylation contexts in this code path previously.